### ISSUE TITLE
MethodError caused by MOI.set(model, MOI.ConstraintDualStart(), [...])

### ISSUE THREAD

**Issue by migarstka** [JuliaOpt/MathOptInterface.jl#955]
I get an error when trying to provide a vector of warm starting variables for `ConstraintDualStart()` 
and calling `allocate_load()`
**MWE:**
```julia
using MathOptInterface, SCS, Test, LinearAlgebra
const MOI = MathOptInterface
const MOIT = MOI.Test
const MOIB = MOI.Bridges
const MOIU = MOI.Utilities
MOIU.@model(ModelData,
        (),
        (MOI.EqualTo, MOI.GreaterThan, MOI.LessThan, MOI.Interval),
        (MOI.Zeros, MOI.Nonnegatives, MOI.Nonpositives, MOI.SecondOrderCone,
         MOI.PositiveSemidefiniteConeSquare, MOI.PositiveSemidefiniteConeTriangle, MOI.ExponentialCone, MOI.DualExponentialCone),
        (MOI.PowerCone, MOI.DualPowerCone),
        (),
        (MOI.ScalarAffineFunction,),
        (MOI.VectorOfVariables,),
        (MOI.VectorAffineFunction,),);

c = [-1.; -1.]    
model = MOIU.UniversalFallback(ModelData{Float64}());
optimizer =  SCS.Optimizer();

x = MOI.add_variables(model, 2);
objectiveFunction = MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.(c, x[1:2]),0.0);
MOI.set(model, MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}(), objectiveFunction);
MOI.set(model, MOI.ObjectiveSense(), MOI.MIN_SENSE);

con1 = MOI.add_constraint(model, MOI.VectorOfVariables(x[1:2]), MOI.Nonnegatives(2));
y_c1 = rand(2)
# warm start the dual variable
MOI.set(model, MOI.ConstraintDualStart(), con1, y_c1)
MOI.empty!(optimizer);
MOIU.allocate_load(optimizer, model, false)
```

This produces the following error in `map_indices` (in `_pass_attributes`) because `value` is of type `Array{Array{Float64,1}}`:

**Error:**
```julia
ERROR: MethodError: no method matching map_indices(::getfield(MathOptInterface.Utilities, Symbol("##7#8")){MathOptInterface.Utilities.IndexMap}, ::Array{Array{Float64,1},1})
Closest candidates are:
  map_indices(::AbstractDict{T<:Union{MathOptInterface.VariableIndex, MathOptInterface.ConstraintIndex},T<:Union{MathOptInterface.VariableIndex, MathOptInterface.ConstraintIndex}}, ::Any) where T<:Union{MathOptInterface.VariableIndex, MathOptInterface.ConstraintIndex} at /Users/Micha/.julia/dev/MathOptInterface/src/Utilities/functions.jl:73
  map_indices(::Function, ::Union{Nothing, AbstractString, DataType, Function, Number, Tuple{Vararg{Union{Nothing, AbstractString, DataType, Function, Number, MathOptInterface.AbstractConstraintAttribute, MathOptInterface.AbstractModelAttribute, MathOptInterface.AbstractOptimizerAttribute, MathOptInterface.AbstractSet, MathOptInterface.AbstractVariableAttribute, Enum},N} where N}, MathOptInterface.AbstractConstraintAttribute, MathOptInterface.AbstractModelAttribute, MathOptInterface.AbstractOptimizerAttribute, MathOptInterface.AbstractSet, MathOptInterface.AbstractVariableAttribute, AbstractArray{#s48,N} where N where #s48<:Union{Nothing, AbstractString, DataType, Function, Number, Tuple{Vararg{Union{Nothing, AbstractString, DataType, Function, Number, MathOptInterface.AbstractConstraintAttribute, MathOptInterface.AbstractModelAttribute, MathOptInterface.AbstractOptimizerAttribute, MathOptInterface.AbstractSet, MathOptInterface.AbstractVariableAttribute, Enum},N} where N}, MathOptInterface.AbstractConstraintAttribute, MathOptInterface.AbstractModelAttribute, MathOptInterface.AbstractOptimizerAttribute, MathOptInterface.AbstractSet, MathOptInterface.AbstractVariableAttribute, Enum}, Enum}) at /Users/Micha/.julia/dev/MathOptInterface/src/Utilities/functions.jl:80
  map_indices(::Function, ::MathOptInterface.VariableIndex) at /Users/Micha/.julia/dev/MathOptInterface/src/Utilities/functions.jl:82
  ...
Stacktrace:
 [1] map_indices(::MathOptInterface.Utilities.IndexMap, ::Array{Array{Float64,1},1}) at /Users/Micha/.julia/dev/MathOptInterface/src/Utilities/functions.jl:73
 [2] _pass_attributes(::SCS.Optimizer, ::MathOptInterface.Utilities.UniversalFallback{COSMOModelData{Float64}}, ::Bool, ::MathOptInterface.Utilities.IndexMap, ::Array{MathOptInterface.AbstractConstraintAttribute,1}, ::Tuple{DataType}, ::Tuple{Array{MathOptInterface.ConstraintIndex{MathOptInterface.VectorOfVariables,MathOptInterface.Nonnegatives},1}}, ::Tuple{Array{MathOptInterface.ConstraintIndex{MathOptInterface.VectorOfVariables,MathOptInterface.Nonnegatives},1}}, ::typeof(MathOptInterface.Utilities.allocate)) at /Users/Micha/.julia/dev/MathOptInterface/src/Utilities/copy.jl:147
 [3] pass_attributes(::SCS.Optimizer, ::MathOptInterface.Utilities.UniversalFallback{COSMOModelData{Float64}}, ::Bool, ::MathOptInterface.Utilities.IndexMap, ::Array{MathOptInterface.ConstraintIndex{MathOptInterface.VectorOfVariables,MathOptInterface.Nonnegatives},1}, ::Function) at /Users/Micha/.julia/dev/MathOptInterface/src/Utilities/copy.jl:124
 [4] pass_constraints(::SCS.Optimizer, ::MathOptInterface.Utilities.UniversalFallback{COSMOModelData{Float64}}, ::Bool, ::MathOptInterface.Utilities.IndexMap, ::Array{DataType,1}, ::Array{Array{#s112,1} where #s112<:(MathOptInterface.ConstraintIndex{MathOptInterface.SingleVariable,S} where S),1}, ::Array{DataType,1}, ::Array{Array{#s13,1} where #s13<:(MathOptInterface.ConstraintIndex{MathOptInterface.VectorOfVariables,S} where S),1}, ::typeof(MathOptInterface.Utilities.allocate_constraints), ::typeof(MathOptInterface.Utilities.allocate)) at /Users/Micha/.julia/dev/MathOptInterface/src/Utilities/copy.jl:255
 [5] allocate_load(::SCS.Optimizer, ::MathOptInterface.Utilities.UniversalFallback{COSMOModelData{Float64}}, ::Bool) at /Users/Micha/.julia/dev/MathOptInterface/src/Utilities/copy.jl:684
 [6] top-level scope at Enums.jl:0
```

- It doesn't seem to me like calling `map_indices` is necessary in the case of a `ConstraintDualStart` attribute. Would defining a method, that does nothing in that case, work?
- I couldn't find any unit tests that use `MOI.set(model, MOI.ConstraintDualStart(), ...)`

**Comment by blegat** [comment]
Should have been fixed by https://github.com/JuliaOpt/MathOptInterface.jl/pull/941 which was included in MOI v0.9.7

**Comment by migarstka** [comment]
Yes, updating from `0.9.5` to `0.9.7` solved that problem. I guess finding the bug twice is better than not finding the bug
